quinn-proto/src/connection/spaces.rs

```diff
@@ -302,24 +302,12 @@ impl Dedup {
     }
 }
 
-pub struct CryptoSpace<K>
+pub struct CryptoSpace<S>
 where
-    K: crypto::Keys,
+    S: crypto::Session,
 {
-    pub packet: K,
-    pub header: K::HeaderKeys,
-}
-
-impl<K> CryptoSpace<K>
-where
-    K: crypto::Keys,
-{
-    pub fn new(packet: K) -> Self {
-        Self {
-            header: packet.header_keys(),
-            packet,
-        }
-    }
+    pub packet: crypto::KeyPair<S::Key>,
```

Comment:
(Also maybe consider importing `KeyPair` directly rather than prefixing with `crypto`? I think it was like this for `Session` because that word seems less obviously crypto-related, but that problem doesn't seek as big of an issue for `KeyPair`... and actually maybe it's also not an issue for `Session` in our code base?)

Reply:
I went back and forth on this a few times, but [the TLS draft](https://quicwg.org/base-drafts/draft-ietf-quic-tls.html) consistently uses the term "packet protection key," and I think that wins out.

Reply:
Fair enough -- I've instead brought it up in quicdev to see if we should have it renamed in the draft.